src/Renderer/MaterialLayer.js

```diff
@@ -126,22 +130,30 @@ class MaterialLayer {
             }
         }
         this.level = EMPTY_TEXTURE_ZOOM;
-        this.textures = [];
-        this.offsetScales = [];
+        this.textures.forEach((v, k) => this.textures.set(k, null));
+        this.offsetScales.forEach((v, k) => this.offsetScales.set(k, null));
         this.material.layersNeedUpdate = true;
     }
 
-    setTexture(index, texture, offsetScale) {
-        this.level = (texture && (index == 0)) ? texture.coords.zoom : this.level;
-        this.textures[index] = texture || null;
-        this.offsetScales[index] = offsetScale;
+    setTexture(texture, extent) {
+        this.level = texture ? texture.coords.zoom : this.level;
+        this.textures.set(extent, texture || null);
```

Comment:
I'm not a fan of using `extent` as a key, maybe it's me but using such a "complex" key when getting something in a `Map` seems to be slow and overkill. Why not write a method that generate a unique key from the content of the extent ?

However, I see two drawbacks in this suggestion (that needs to be verified):
- the conversion to a string can mean a lot of process if it is called a lot
- I'm not sure if the key is used as it is sometimes (it seems in `updateLayeredMaterialNodeImagery`, so...)

Reply:
Me too, I'm not fan. We can optimize later.

Reply:
Good to me